compiler/rustc_passes/src/dead.rs

```diff
@@ -266,6 +284,17 @@ impl<'tcx> MarkSymbolVisitor<'tcx> {
                         intravisit::walk_item(self, &item);
                     }
                     hir::ItemKind::ForeignMod { .. } => {}
+                    hir::ItemKind::Impl(_) => {
+                        if self.should_ignore_impl(item.def_id.to_def_id()) {
+                            // We are looking at an implementation of `Clone` or `Debug` that was
+                            // automatically derived via #[derive(...)]. Ignore this for dead code
+                            // analysis (derived implementations of Debug and Clone always trivially
+                            // use all fields, so they would hide unused field warnings for such
+                            // structs/enums, cf. issue #84647).
```

Comment:
Maybe just move this comment and change this code to `if !self.should_ignore_impl(...) { .. }`